scripts/pylib/pytest-twister-harness/src/twister_harness/helpers/shell.py

```diff
@@ -65,4 +66,8 @@ def exec_command(self, command: str, timeout: float | None = None, print_output:
         lines.extend(self._device.readlines_until(regex=regex_command, timeout=1.0, print_output=print_output))
         # wait for device command execution
         lines.extend(self._device.readlines_until(regex=regex_prompt, timeout=timeout, print_output=print_output))
-        return lines
+        if filtered:
+            r = re.compile(regex_command + '|' + regex_prompt + '|<dbg>|<inf>|<wrn>|<err>')
+            return list(filter(lambda l: not r.search(l), lines))
+        else:
+            return lines
```

Comment:
```suggestion
    def get_filtered_output(self, command_lines: list[str]) -> list[str]:
        regex_filter = re.compile(
            '|'.join(
                re.escape(self.prompt),
                '<dbg>',
                '<inf>',
                '<wrn>',
                '<err>'
            )
        )
        return list(filter(lambda l: not regex_filter.search(l), command_lines))
```
Consider adding new method to filter output, instead of adding extra parameters and if-statements.